tiledb/sm/query/deletes_and_updates/deletes_and_updates.cc

```diff
@@ -163,13 +165,14 @@ Status DeletesAndUpdates::dowork() {
 
   // Serialize the negated condition (aud update values if they are not empty)
   // and write to disk.
+  QueryCondition qc(condition_.value_or(QueryCondition()));
```

Comment:
I'm not sure what you mean by the constructor guaranteeing a condition here. As for the value() throwing, I'm using `value_or()` to return either the value that's present or the passed default.

Reply:
Yeah I guess the serialization path in the constructor bypasses the guarantee. Either way we should throw at the top of dowork if there's no value in the condition. That way the rest of the function can assume a condition. What I'm also trying to prevent with value_or is that we use the default constructor for condition and eventually we'll get rid of the default constructor to make a C.41 object.

Reply:
I still don't think we're on the same page here. My confusion about the constructor guaranteeing a condition was that its a `std::optional<QueryCondition>` now, so a `std::nullopt` is perfectly reasonable to guard against. I haven't trolled very deep into the serialization logic, so I could see changing those functions to accept the optional and then do checking further down the call stack if that's what you mean. Though that would move the negation logic possible which seems wrong.

Reply:
Also, I do see your point about eventually removing a default `QueryCondition` constructor making `value_or` something we'd have to fix later. I just have no idea what we'd expect to pass to the serialization calls on lines 172 and 175 if `condition_` is `std::nullopt`.

Reply:
Sorry for the confusion... Let's scratch my comment about the constructor, it wasn't correct.

I think at the top of dowork(), we add:
`if (!condition_.has_value()) {`
`  throw DeleteAndUpdateStatusException("Cannot process delete, no condition is set");`
`}`

Then in the serialization codepath, you can just pass condition_->negated_condition(). We can get rid of the qc variable altogether.

Reply:
Got this straightened out on a call. We know that when `dowork()` is called for updates and deletes that we have a query condition available and its an error otherwise so throwing there is fine. The reason we can't do it in the constructor is that the current serialization implementation constructs objects and then updates their state. So in that case we will have a constructed instance with a `std::nullopt` stored in `condition_`.

Reply:
Fixed.